File: 938. Range Sum of BST/Range Sum of BST.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def rangeSumBST(self, root: Optional[TreeNode], low: int, high: int) -> int:
        check = [root]
        result = 0
        while len(check) > 0:
            current = check.pop()
            if current.val < low or current.val > high:
                pass
            else:
                result += current.val
            if current.left != None:
                check.append(current.left)
            if current.right != None:
                check.append(current.right)
        return result

File: 938. Range Sum of BST/Range Sum of BST.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int rangeSumBST(TreeNode root, int low, int high) {
        ArrayList<TreeNode> check = new ArrayList<TreeNode>();
        int result = 0;
        check.add(root);
        while (!check.isEmpty()){
            TreeNode current = check.remove(check.size()-1);
            if (current.val >= low && current.val <= high){
                result += current.val;
            }
            if (current.left != null){
                check.add(current.left);
            }
            if (current.right != null){
                check.add(current.right);
            }
        }
        return result;
    }
}
